pgtype/interval.go

```diff
@@ -26,6 +26,21 @@ type IntervalValuer interface {
 	IntervalValue() (Interval, error)
 }
 
+// Interval represents an interval that may be null.
+// Interval implements the [Scanner] interface so
+// it can be used as a scan destination:
```

Comment:
Implementing Scanner is not why it can be used as a destination in pgx. That's only when used through database/sql.